src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -235,180 +276,215 @@ protected override AuthenticationTicket AuthenticateCore()
             {
                 IFormCollection form = await Request.ReadFormAsync();
                 Request.Body.Seek(0, SeekOrigin.Begin);
-
-                openIdConnectMessage = new OpenIdConnectMessage(form);
+                message = new OpenIdConnectMessage(form);
             }
 
-            if (openIdConnectMessage == null)
+            if (message == null)
```

Comment:
Security considerations are already listed on the OAuth2 multiple response_mode specs (http://openid.net/specs/oauth-v2-multiple-response-types-1_0.html):

> There are security implications to encoding response values in the query string. The HTTP Referer header includes query parameters, and so any values encoded in query parameters will leak to third parties. Thus, while it is safe to encode an Authorization Code as a query parameter when using a Confidential Client (because it can't be used without the Client Secret, which third parties won't have), more sensitive information such as Access Tokens and ID Tokens MUST NOT be encoded in the query string. In no case should a set of Authorization Response parameters whose default Response Mode is the fragment encoding be encoded using the query encoding. 

As mentioned, supporting `response_mode=query` + `response_type=code` is safe. If it was not the case, `query` wouldn't be the default `response_mode` for `response_type=code`. 

Blocking GET callbacks exposing `token` or `id_token` would be enough to make sure safe `response_type`/`response_mode` combinations are used.

(related topic: https://katanaproject.codeplex.com/SourceControl/network/forks/manfredsteyer/OIDCResponseModeQuery/contribution/7009)

/cc @selfissued